packages/editor/src/lib/primitives/Vec.ts

```diff
@@ -594,4 +595,36 @@ export class Vec {
 	}
 }
 
+/**
+ * A {@link Vec} backed by atoms. When the atom changes, anything reading its values will re-evaluate.
+ * @public
+ */
+export class AtomVec extends Vec {
+	constructor(name: string, x = 0, y = 0, z = 1) {
+		super(x, y, z)
+		this._x = atom(`${name}.x`, x)
+		this._y = atom(`${name}.y`, y)
+		this._z = atom(`${name}.z`, z)
+
+		Object.defineProperties(this, {
+			x: {
+				get: () => this._x.get(),
+				set: (value: number) => this._x.set(value),
+			},
+			y: {
+				get: () => this._y.get(),
+				set: (value: number) => this._y.set(value),
+			},
+			z: {
+				get: () => this._z.get(),
+				set: (value: number) => this._z.set(value),
+			},
+		})
+	}
+
+	private _x: Atom<number>
+	private _y: Atom<number>
+	private _z: Atom<number>
+}
```

Comment:
one of our big customers uses terser in their build pipeline with the `pure_getters` setting enabled, which prevents signals from being tracked in lots of cases. Which is why we originally removed all signal-capturing getters from our codebase a long while back.

I'm not sure if defining the setters like this would cause the same problem, but we definitely need to check and address it if so.

Reply:
> Does that match your understanding? 

I vaguely remember it being more complicated than that, like our customer had a combination of terser settings (or maybe terser + some other compiler?) that meant it would unsafely memoize the getter value sometimes. I could be wrong, it was a long time ago.

> I think the alternative is making each of these into a function which is obviously a much bigger change from an api point of view, but maybe that's the way to go here?

Yeah maybe. The alternative is to try it like this and cross fingers that our customer doesn't notice any bugs, but that doesn't feel like a winning strategy to me 😵‍💫